### PR TITLE
Remove The Implicit Type Filter

### PR DESCRIPTION
**Pull request by fppt**
This is the best PR ever made. The greatest even. No one will ever beat this PR!

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/session/GraqlSession.java [hunk 1]
```diff
@@ -122,7 +122,6 @@
     private void refreshGraph() {
         if (graph != null && !graph.isClosed()) graph.close();
         graph = factory.open(GraknTxType.WRITE);
-        graph.showImplicitConcepts(showImplicitTypes);
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/session/GraqlSession.java]: There's still a variable `showImplicitTypes` here. We need to kill it and all it's friends and family (that means the flag in `GraqlShell`).

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryBase.java [hunk 2]
```diff
@@ -179,24 +178,6 @@ public final MatchQuery infer(boolean materialise) {
...
-
-    /**
-     * Only show a concept if it not an implicit type and not explicitly mentioned
-     */
-    private boolean shouldShowConcept(GraknGraph graph, Concept concept) {
-        if (graph.implicitConceptsVisible() || !concept.isType()) return true;
-
-        Type type = concept.asType();
-
-        return !type.isImplicit() || labels.contains(type.getLabel());
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryBase.java]: It's possible we're not using the `labels` field anymore.

File: grakn-test-tools/src/main/java/ai/grakn/generator/GraknGraphs.java [hunk 3]
```diff
@@ -184,7 +182,6 @@ public GraknGraphs setOpen(boolean open) {
             },
             () -> {
                 boolean flag = gen(Boolean.class);
-                graph.showImplicitConcepts(flag);
```
Comment (aelred) [inline grakn-test-tools/src/main/java/ai/grakn/generator/GraknGraphs.java]: Kill this lambda entirely. It does nothing now!

**Comment by fppt** [comment]
@aelred re-review please

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryBase.java [hunk 4]
```diff
@@ -84,8 +83,6 @@ public MatchQueryBase(Conjunction<PatternAdmin> pattern) {
     public Stream<Answer> stream(Optional<GraknGraph> optionalGraph) {
         GraknGraph graph = optionalGraph.orElseThrow(GraqlQueryException::noGraph);
 
-        this.labels = getAllTypeLabels(graph);
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryBase.java]: You can also kill this method 🔪

File: grakn-engine/src/main/java/ai/grakn/engine/session/RemoteSession.java [hunk 5]
```diff
@@ -97,12 +97,11 @@ private void startSession(Json json) {
         if (sessionAuthorised(json)) {
             String keyspace = json.at(REST.RemoteShell.KEYSPACE).asString();
             String outputFormat = json.at(REST.RemoteShell.OUTPUT_FORMAT).asString();
-            boolean showImplicitTypes = json.at(REST.RemoteShell.IMPLICIT).asBoolean();
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/session/RemoteSession.java]: Now you can kill it on the client-side as well (that's class `GraqlShell`), plus the docs.

**Comment by fppt** [comment]
@aelred re-re-review please. 3rds the charm. Thanks for reminding me about the docs!

**Comment by aelred** [comment]
This PR is just waiting on a fix to dashboard @marco-scoppetta 
image

**Comment by fppt** [comment]
@aelred the fixed by @marco-scoppetta are in. re-re-re-review please 😄

**Pull request merged**